pkg/runtime/interfaces.go

```diff
@@ -45,6 +45,11 @@ type ObjectTyper interface {
 	ObjectVersionAndKind(Object) (version, kind string, err error)
 }
 
+// ObjectCreater contains methods for instantiating an object by kind and version.
+type ObjectCreater interface {
```

Comment:
I mostly did this to be consistent with ObjectTyper.  Defer until we clean up runtime?